power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -33,21 +33,30 @@ template <component_c Component, class ComponentContainer,
           std::output_iterator<Idx2D> OutputIterator>
     requires model_component_state_c<MainModelState, ComponentContainer, Component>
 inline void get_component_sequence(MainModelState<ComponentContainer> const& state, ForwardIterator begin,
-                                   ForwardIterator end, OutputIterator destination) {
+                                   ForwardIterator end, OutputIterator destination, Idx n_comp_elements) {
```

Comment:
It the number of entries in corresponding component, since the `components` contains a list of them
update data
-- scenario
----component
------elements